### PR TITLE
[Hotfix][ENG-3205] Add Character Lab Long and Short Form Registration Schemas

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Give CL two new schemas!

## Changes

- adds two new schemas for CL

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-3205

### CODE REVIEW

File: website/project/metadata/character-lab-long-form-registration.json [hunk 1]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "These outcomes were developed by the Impact Genome Project \u00ae and are licensed under a Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License. To learn more, go to impactgenome.org",
+		"block_type": "question-label",
+		"display_text": "Expected outcomes",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:32]: should be "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 2]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Character strengths",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:194]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 3]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "These outcomes were developed by the Impact Genome Project \u00ae and are licensed under a Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License. To learn more, go to impactgenome.org",
+		"block_type": "question-label",
+		"display_text": "Expected outcomes",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:32]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 4]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Character strengths",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:194]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 5]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Study Grade Assignment",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:398]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 6]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "School Stratum Assignment",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:452]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 7]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Study design",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:524]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 8]
```diff
@@ -0,0 +1,1021 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Team member role",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:608]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 9]
```diff
@@ -0,0 +1,1021 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:698]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 10]
```diff
@@ -0,0 +1,1021 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:788]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 11]
```diff
@@ -0,0 +1,1021 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:878]: -> "multi-select-input"

File: website/project/metadata/character-lab-short-form-registration.json [hunk 12]
```diff
@@ -0,0 +1,1021 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-short-form-registration.json:968]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 13]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Study Grade Assignment",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:398]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 14]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "School Stratum Assignment",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:452]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 15]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Study design",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:524]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 16]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Team member role",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:608]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 17]
```diff
@@ -0,0 +1,1195 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:698]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 18]
```diff
@@ -0,0 +1,1195 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:788]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 19]
```diff
@@ -0,0 +1,1195 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:878]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 20]
```diff
@@ -0,0 +1,1195 @@
...
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:968]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 21]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Have you piloted this study?",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:1034]: -> "multi-select-input"

File: website/project/metadata/character-lab-long-form-registration.json [hunk 22]
```diff
@@ -0,0 +1,1195 @@
...
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "question-label",
+		"display_text": "Did you pilot this study on the Character Lab Research Network?",
+		"example_text": "",
+		"required": false
+	}, {
+		"help_text": "",
+		"block_type": "multiple-input-option",
```
Comment (jwalz) [inline website/project/metadata/character-lab-long-form-registration.json:1058]: -> "multi-select-input"

**Review by jwalz — CHANGES_REQUESTED** [review]
There's a discrepancy between the CSV and the actual schema block types. They use "multiple-input-option" instead of "multi-select-input". I flagged all of the spots where we got caught by that.

Other than that, it sounds like this should now be merged to Develop instead of Hotfixed, in which case you could also incorporate some of the readability improvements from https://github.com/CenterForOpenScience/osf.io/pull/9747 (e.g. not including empty help_text/example_text)

**Review by jwalz — CHANGES_REQUESTED** [review]
still using the unsupported `multiple-select-option` value for `block_type` instead of `multi-select-input`, also, trailing commas are breaking json validation. I fixed the spreadsheet values with the wrong `block_type`, if that helps.

**Comment by jwalz** [comment]
Oh, and these files need to be added to the list in website/project/metadata/schemas.py

File: website/project/metadata/character-lab-long-form-registration.json [hunk 23]
```diff
@@ -0,0 +1,565 @@
...
+		"display_text": "Stratum D: Mostly Black, high # of students receiving free/reduced lunch, often urban schools. These schools represent 12% of all middle and high schools in the US."
+	}, {
+		"block_type": "select-input-option",
+		"display_text": "Stratum E: Mostly White, high socioeconomic status, suburban schools. These schools represent 21% of all middle and high schools in the United States."
+	}, {
+		"block_type": "page-heading",
+		"display_text": "Study Information"
+	}, {
+		"block_type": "question-label",
+		"display_text": "Is this a pilot?"
```
Comment (adlius) [inline website/project/metadata/character-lab-long-form-registration.json:244]: there should be a `single-select-input` block after this block.

Comment (adlius) [inline website/project/metadata/character-lab-long-form-registration.json:244]: There should be a `single-select-input` block after this block.

**Review by adlius — CHANGES_REQUESTED** [review]

**Comment by jwalz** [comment]
Closed in favor of https://github.com/CenterForOpenScience/osf.io/pull/9767

**Pull request closed without merging**